### PR TITLE
docs: revised the online-offline event detection feature page

### PR DESCRIPTION
**Pull request by bandantonio**
#### Description of Change
Revised the online-offline event detection feature page, reworked code samples, added screenshot that corresponds to the code sample

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Revised the online-offline event detection feature page, reworked code samples, added screenshot that corresponds to the code sample

### CODE REVIEW

**Review by zcbenz — CHANGES_REQUESTED** [review]
We are going to add an API to return online status in the main process, I think we can use the new API in this doc and merge after the API gets merged.

https://github.com/electron/electron/pull/21004

File: docs/tutorial/online-offline-events.md
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:24]:
```suggestion
### Event detection in the Renderer process
```

File: docs/tutorial/online-offline-events.md
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:63]:
```suggestion
### Event detection in the Main process
```

File: docs/tutorial/online-offline-events.md
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:12]:
```suggestion
The `navigator.onLine` attribute returns:
* `false` if all network requests are guaranteed to fail (e.g. when disconnected
from the network).
```

**Review by erickzhao — COMMENTED** [review]
Comments for change.

@zcbenz: does this mean that we should wait for the next Electron release to land in stable? I think `/docs/guides/` fetches its data from `master`, so the `net.online` API won't be usable for most users for a while after it lands.

**Comment by zcbenz** [comment]
> @zcbenz: does this mean that we should wait for the next Electron release to land in stable? I think `/docs/guides/` fetches its data from `master`, so the `net.online` API won't be usable for most users for a while after it lands.

Hmm I am fine landing current document first and then update it after the new API is stable, it would require some extra work though.

**Comment by erickzhao** [comment]
Sounds good to me. @bandantonio's Season of Docs term is only until December, so let's ship faster and revisit later.

**Review by erickzhao — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**
